### PR TITLE
Replicas max per node

### PR DESCRIPTION
**Pull request by WojciechowskiPiotr**
##### SUMMARY
Adding support for MaxReplicas Placement parameter for `docker_swarm_service` (requested in https://github.com/ansible-collections/community.docker/issues/7)

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
docker_swarm_service

### CODE REVIEW

File: plugins/modules/docker_swarm_service.py [hunk 1]
```diff
@@ -344,6 +345,12 @@
           - Requires API version >= 1.27.
         type: list
         elements: dict
+      replicas_max_per_node:
+        description:
+          - Maximum number of tasks per node.
+          - Corresponds to the C(--replicas_max_per_node) option of C(docker service create).
+          - Requires API version >= 1.40
+        type: int
```
Comment (felixfontein) [inline plugins/modules/docker_swarm_service.py:353]:
```suggestion
        type: int
        version_added: 1.3.0
```

**Review by felixfontein — COMMENTED** [review]
I think you also need to update the return value documentation, since you added a new field. (I think that new field is also what makes the tests fail.)

File: plugins/modules/docker_swarm_service.py (hunk 1 above)
Comment (WojciechowskiPiotr) [reply plugins/modules/docker_swarm_service.py:353]: Let's see if this is the problem. Iinteresting that only tests on two versions of RHEL failed.

Comment (felixfontein) [reply plugins/modules/docker_swarm_service.py:353]: The tests probably only run on that two nodes. (All swarm-related tests don't run in the docker containers, only on VMs.)

**Comment by WojciechowskiPiotr** [comment]
ready_for_review

**Review by felixfontein — COMMENTED** [review]
You should also mention `replicas_max_per_node` in the return example (https://github.com/ansible-collections/community.docker/blob/8f209e4070f82c4d0f9c0b876a952b5fa4d0ad17/plugins/modules/docker_swarm_service.py#L728). Also this needs a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).

Besides that, looks good to me. As far as I can judge changes to this module :)

@hannseman @jwitko @dariko any comments on this?

**Review by hannseman — COMMENTED** [review]
Looks good to me! 👍

**Comment by WojciechowskiPiotr** [comment]
> You should also mention `replicas_max_per_node` in the return example (
> 
> https://github.com/ansible-collections/community.docker/blob/8f209e4070f82c4d0f9c0b876a952b5fa4d0ad17/plugins/modules/docker_swarm_service.py#L728
> 
> ). Also this needs a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to).
> Besides that, looks good to me. As far as I can judge changes to this module :)
> 
> @hannseman @jwitko @dariko any comments on this?

Updated

File: changelogs/fragments/92-replicas-max-per-node.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - "docker_swarm_service - Adding support for maximum number of tasks per node (replicas_max_per_node) when running swarm service in replicated mode. Introduced in API 1.40"
```
Comment (felixfontein) [inline changelogs/fragments/92-replicas-max-per-node.yml:2]:
```suggestion
  - "docker_swarm_service - adding support for maximum number of tasks per node (``replicas_max_per_node``) when running swarm service in replicated mode. Introduced in API 1.40 (https://github.com/ansible-collections/community.docker/issues/7, https://github.com/ansible-collections/community.docker/pull/92)."
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@WojciechowskiPiotr thanks for implementing this!
@hannseman thanks for reviewing!

If you have some time, can you take a look at #87, #89, #90? :) I'll also try to create a PR for #91 today, I think with that we can release 1.3.0 very soon.

**Pull request merged**